src/util/BinaryFuseFilter.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+// Copyright 2024 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "lib/binaryfusefilter.h"
+#include "util/NonCopyable.h"
+#include "xdr/Stellar-ledger-entries.h"
+#include "xdr/Stellar-types.h"
+
+#include <cereal/archives/binary.hpp>
+#include <xdrpp/xdrpp/cereal.h>
+
+namespace stellar
+{
+
+// This class is a wrapper around the binary_fuse_t library that provides
+// serialization for the XDR BinaryFuseFilter type and provides a deterministic
+// LedgerKey interface.
+
+// Only allow uint8_t, uint16_t, and uint32_t
+template <typename T,
+          typename = std::enable_if_t<std::is_unsigned<T>::value &&
+                                      !std::is_same<T, uint64_t>::value>>
+class BinaryFuseFilter : public NonMovableOrCopyable
+{
+  private:
+    binary_fuse_t<T> mFilter;
+
+    // Note: as part of filter construction, the internal filter seed might
+    // rotate and no longer be the same as the input seed. The input seed must
+    // be maintained outside of the filter and used to hash input keys in the
+    // contain function to ensure deterministic hashing of input keys
+    // during both populating and querying the filter.
+    binary_fuse_seed_t const mInputSeed;
```

Comment:
Assuming I understand the difference here, I suggest a change to the comment and variable name.

```suggestion
    // Note: This is the seed used to hash keys going into the filter, and we need to preserve it
    // to hash keys we're looking up in the filter. The filter also has its own seed determining its
    // overall structure, which _starts_ with this seed, but which may be rotated a bit in during
    // repeated attempts to successfully populate the filter. So we have to keep both seeds. 
    binary_fuse_seed_t const mHashSeed;
```